test_scripts/API/SetAppIcon/commonIconResumed.lua

```diff
@@ -0,0 +1,140 @@
+---------------------------------------------------------------------------------------------------
+-- Common module
+---------------------------------------------------------------------------------------------------
+---------------------------------------------------------------------------------------------------
+--[[ General configuration parameters ]]
+config.defaultProtocolVersion = 2
+
+--[[ Required Shared libraries ]]
+local actions = require("user_modules/sequences/actions")
+local utils = require("user_modules/utils")
+local test = require("user_modules/dummy_connecttest")
+local commonPreconditions = require('user_modules/shared_testcases/commonPreconditions')
+
+--[[ Module ]]
+local m = actions
+
+--[[ Variables ]]
+-- local ptuTable = {}
+local hmiAppIds = {}
+
+function m.getPathToFileInStorage(fileName, pAppId)
+  if not pAppId then pAppId = 1 end
+  return commonPreconditions:GetPathToSDL() .. "storage/"
+  .. m.getConfigAppParams( pAppId ).appID .. "_"
+  .. utils.getDeviceMAC() .. "/" .. fileName
+end
+
+--[[ @registerAppWOPTU: register mobile application
+--! @parameters:
+--! pAppId - application number (1, 2, etc.)
+--! pIconResumed - apps icon was resumed at system or is not resumed
+--! pReconnection - re-register mobile application
+--! @return: none
+--]]
+function m.registerAppWOPTU(pAppId, pIconResumed, pReconnection, pIconValue)
+  if not pAppId then pAppId = 1 end
+  if pIconResumed == true then
+    if not pIconValue then
+      pIconValue = m.getPathToFileInStorage("icon.png")
+    else
+      pIconValue = m.getPathToFileInStorage(pIconValue)
+    end
+  end
+  local mobSession = m.getMobileSession(pAppId)
+  local function RegisterApp()
+    local corId = mobSession:SendRPC("RegisterAppInterface",
+        config["application" .. pAppId].registerAppInterfaceParams)
+      test.hmiConnection:ExpectNotification("BasicCommunication.OnAppRegistered",
+        { application = {
+            appName = config["application" .. pAppId].registerAppInterfaceParams.appName,
+            icon = pIconValue
+        }})
+      :Do(function(_, d1)
+          hmiAppIds[m.getConfigAppParams(pAppId).appID] = d1.params.application.appID
+        end)
+      :ValidIf(function(_,data)
+        if false == pIconResumed and
+          data.params.application.icon then
+          return false, "BC.OnAppRegistered notification contains unexpected icon value "
```

Comment:
@HSavynetska please specify in error what icon value was expected and what was received

Reply:
Please, rephrase : 
BC.OnAppRegistered notification contains unexpected parameter : icon